From 06041cc74724f2cc59cd3c750cd5846ab0055d52 Mon Sep 17 00:00:00 2001
From: Tong Wu <tong1.wu@intel.com>
Date: Mon, 17 Oct 2022 16:48:15 +0800
Subject: [PATCH 05/16] msdkallocator: add a parameter to function
 gst_msdk_import_to_msdk_surface

Add a flags parameter to function gst_msdk_import_to_msdk_surface and
move the function declaration to gstmsdkallcator.h to prepare for adding
Windows implementation.
---
 subprojects/gst-plugins-bad/sys/msdk/gstmsdkallocator.h   | 4 ++++
 .../gst-plugins-bad/sys/msdk/gstmsdkallocator_libva.c     | 5 +++--
 .../gst-plugins-bad/sys/msdk/gstmsdkallocator_libva.h     | 4 ----
 subprojects/gst-plugins-bad/sys/msdk/gstmsdkenc.c         | 4 ++--
 subprojects/gst-plugins-bad/sys/msdk/gstmsdkvpp.c         | 8 ++++----
 5 files changed, 13 insertions(+), 12 deletions(-)

diff --git a/subprojects/gst-plugins-bad/sys/msdk/gstmsdkallocator.h b/subprojects/gst-plugins-bad/sys/msdk/gstmsdkallocator.h
index b8d905095b..3435048a0b 100644
--- a/subprojects/gst-plugins-bad/sys/msdk/gstmsdkallocator.h
+++ b/subprojects/gst-plugins-bad/sys/msdk/gstmsdkallocator.h
@@ -78,6 +78,10 @@ void gst_msdk_set_frame_allocator (GstMsdkContext * context);
 GQuark
 gst_msdk_frame_surface_quark_get (void);
 
+GstMsdkSurface *
+gst_msdk_import_to_msdk_surface (GstBuffer * buf, GstMsdkContext * msdk_context,
+    GstVideoInfo * vinfo, guint flags);
+
 G_END_DECLS
 
 #endif /* GST_MSDK_ALLOCATOR_H_ */
diff --git a/subprojects/gst-plugins-bad/sys/msdk/gstmsdkallocator_libva.c b/subprojects/gst-plugins-bad/sys/msdk/gstmsdkallocator_libva.c
index 8acae9f1e6..2d7737e1d6 100644
--- a/subprojects/gst-plugins-bad/sys/msdk/gstmsdkallocator_libva.c
+++ b/subprojects/gst-plugins-bad/sys/msdk/gstmsdkallocator_libva.c
@@ -140,7 +140,7 @@ gst_msdk_frame_alloc (mfxHDL pthis, mfxFrameAllocRequest * req,
         return MFX_ERR_MEMORY_ALLOC;
       }
 
-      msdk_surface = gst_msdk_import_to_msdk_surface (buf, context, &info);
+      msdk_surface = gst_msdk_import_to_msdk_surface (buf, context, &info, 0);
       if (msdk_surface)
         msdk_surface->buf = buf;
       mids[i] = msdk_surface->surface->Data.MemId;
@@ -621,9 +621,10 @@ _get_va_surface (GstBuffer * buf, GstVideoInfo * info,
   return va_surface;
 }
 
+/* Currently parameter flags is not useful on Linux */
 GstMsdkSurface *
 gst_msdk_import_to_msdk_surface (GstBuffer * buf, GstMsdkContext * msdk_context,
-    GstVideoInfo * vinfo)
+    GstVideoInfo * vinfo, guint flags)
 {
   VASurfaceID va_surface = VA_INVALID_ID;
   GstMemory *mem = NULL;
diff --git a/subprojects/gst-plugins-bad/sys/msdk/gstmsdkallocator_libva.h b/subprojects/gst-plugins-bad/sys/msdk/gstmsdkallocator_libva.h
index fa25821fd8..da2f4d3bfe 100644
--- a/subprojects/gst-plugins-bad/sys/msdk/gstmsdkallocator_libva.h
+++ b/subprojects/gst-plugins-bad/sys/msdk/gstmsdkallocator_libva.h
@@ -45,10 +45,6 @@ gboolean
 gst_msdk_export_dmabuf_to_vasurface (GstMsdkContext *context,
     GstVideoInfo *vinfo, gint fd, VASurfaceID *surface_id);
 
-GstMsdkSurface *
-gst_msdk_import_to_msdk_surface (GstBuffer * buf, GstMsdkContext * msdk_context,
-    GstVideoInfo * vinfo);
-
 gboolean
 gst_msdk_replace_mfx_memid (GstMsdkContext *context,
     mfxFrameSurface1 *mfx_surface, VASurfaceID surface_id);
diff --git a/subprojects/gst-plugins-bad/sys/msdk/gstmsdkenc.c b/subprojects/gst-plugins-bad/sys/msdk/gstmsdkenc.c
index 35d93f046e..efa4ed1e18 100644
--- a/subprojects/gst-plugins-bad/sys/msdk/gstmsdkenc.c
+++ b/subprojects/gst-plugins-bad/sys/msdk/gstmsdkenc.c
@@ -1680,7 +1680,7 @@ gst_msdkenc_get_surface_from_pool (GstMsdkEnc * thiz, GstBufferPool * pool,
   }
 #ifndef _WIN32
   msdk_surface = gst_msdk_import_to_msdk_surface (new_buffer, thiz->context,
-      &thiz->aligned_info);
+      &thiz->aligned_info, 0);
 #else
   msdk_surface =
       gst_msdk_import_sys_mem_to_msdk_surface (new_buffer, thiz->aligned_info);
@@ -1708,7 +1708,7 @@ gst_msdkenc_get_surface_from_frame (GstMsdkEnc * thiz,
   }
 #ifndef _WIN32
   msdk_surface = gst_msdk_import_to_msdk_surface (inbuf, thiz->context,
-      &thiz->input_state->info);
+      &thiz->input_state->info, 0);
   if (msdk_surface) {
     msdk_surface->buf = gst_buffer_ref (inbuf);
     return msdk_surface;
diff --git a/subprojects/gst-plugins-bad/sys/msdk/gstmsdkvpp.c b/subprojects/gst-plugins-bad/sys/msdk/gstmsdkvpp.c
index 1b4037a4f9..220e2d2549 100644
--- a/subprojects/gst-plugins-bad/sys/msdk/gstmsdkvpp.c
+++ b/subprojects/gst-plugins-bad/sys/msdk/gstmsdkvpp.c
@@ -734,7 +734,7 @@ get_surface_from_pool (GstMsdkVPP * thiz, GstBufferPool * pool,
   }
 #ifndef _WIN32
   msdk_surface = gst_msdk_import_to_msdk_surface (new_buffer, thiz->context,
-      &thiz->sinkpad_info);
+      &thiz->sinkpad_info, 0);
 #else
   msdk_surface = gst_msdk_import_sys_mem_to_msdk_surface
       (new_buffer, thiz->sinkpad_buffer_pool_info);
@@ -760,7 +760,7 @@ get_msdk_surface_from_input_buffer (GstMsdkVPP * thiz, GstBuffer * inbuf)
   }
 #ifndef _WIN32
   msdk_surface = gst_msdk_import_to_msdk_surface (inbuf, thiz->context,
-      &thiz->sinkpad_info);
+      &thiz->sinkpad_info, 0);
   if (msdk_surface) {
     msdk_surface->buf = gst_buffer_ref (inbuf);
     return msdk_surface;
@@ -846,7 +846,7 @@ gst_msdkvpp_transform (GstBaseTransform * trans, GstBuffer * inbuf,
   } else {
 #ifndef _WIN32
     out_surface = gst_msdk_import_to_msdk_surface (outbuf, thiz->context,
-        &thiz->srcpad_info);
+        &thiz->srcpad_info, 0);
 #else
     out_surface =
         gst_msdk_import_sys_mem_to_msdk_surface (outbuf, thiz->srcpad_info);
@@ -939,7 +939,7 @@ gst_msdkvpp_transform (GstBaseTransform * trans, GstBuffer * inbuf,
         release_out_surface (thiz, out_surface);
 #ifndef _WIN32
         out_surface = gst_msdk_import_to_msdk_surface (outbuf, thiz->context,
-            &thiz->srcpad_info);
+            &thiz->srcpad_info, 0);
 #else
         out_surface =
             gst_msdk_import_sys_mem_to_msdk_surface (outbuf, thiz->srcpad_info);
-- 
2.35.1.windows.2

